[quant][pt2][be] Refactor QAT tests for future patterns

andrewor14 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #113714
* #113709
* __->__ #113658

Summary: Currently the QAT tests are very specific to conv-bn-2d.
This makes it difficult to test new patterns like conv-bn-1d if
we want to add them. This commit refactors these tests so we can
add and test future patterns easily.

Test Plan:
python test/test_quantization.py TestQuantizePT2EQAT_ConvBn2d

Reviewers: jerryzh168, kimishpatel

Subscribers: jerryzh168, kimishpatel, supriyar

andrewor14 committed:
[quant][pt2][be] Refactor QAT tests for future patterns

Summary: Currently the QAT tests are very specific to conv-bn-2d.
This makes it difficult to test new patterns like conv-bn-1d if
we want to add them. This commit refactors these tests so we can
add and test future patterns easily.

Test Plan:
python test/test_quantization.py TestQuantizePT2EQAT_ConvBn2d

Reviewers: jerryzh168, kimishpatel

Subscribers: jerryzh168, kimishpatel, supriyar

[ghstack-poisoned]

andrewor14 committed:
Update on "[quant][pt2][be] Refactor QAT tests for future patterns"


Summary: Currently the QAT tests are very specific to conv-bn-2d.
This makes it difficult to test new patterns like conv-bn-1d if
we want to add them. This commit refactors these tests so we can
add and test future patterns easily.

Test Plan:
python test/test_quantization.py TestQuantizePT2EQAT_ConvBn2d

Reviewers: jerryzh168, kimishpatel

Subscribers: jerryzh168, kimishpatel, supriyar

[ghstack-poisoned]

jerryzh168 commented on test/quantization/pt2e/test_quantize_pt2e_qat.py:
```diff
@@ -304,34 +303,70 @@ def _verify_symmetric_xnnpack_qat_graph_helper(
 
-            def forward(self, x):
-                x = self.conv(x)
+    class _TestModel(torch.nn.Module):
```
how about putting these in `PT2EQATTestCase`?

andrewor14 replied:
sure

andrewor14 commented on test/test_quantization.py:
```diff
@@ -90,7 +90,9 @@
-    from quantization.pt2e.test_quantize_pt2e_qat import TestQuantizePT2EQAT  # noqa: F401
+    # TODO: enable this in the next PR
+    # from quantization.pt2e.test_quantize_pt2e_qat import TestQuantizePT2EQAT_ConvBn1d  # noqa: F401
+    from quantization.pt2e.test_quantize_pt2e_qat import TestQuantizePT2EQAT_ConvBn2d  # noqa: F401
```
I see. `TestQuantizePT2EQAT_ConvBn2d` is actually a group of tests, not a single test. I just felt each pattern already has a ton of tests so it's better to break them down instead of putting them all in one giant TestCase. Do you think we should merge them instead?

jerryzh168 replied:
I think we have multiple ways to structure tests already so ideally want to keep it simpler, we already have different folders, test files, then TestCases and then test names

andrewor14 replied:
Ok, added a TODO. As discussed offline we will merge these test cases into a single one in the future

andrewor14 committed:
Update on "[quant][pt2][be] Refactor QAT tests for future patterns"


Summary: Currently the QAT tests are very specific to conv-bn-2d.
This makes it difficult to test new patterns like conv-bn-1d if
we want to add them. This commit refactors these tests so we can
add and test future patterns easily.

Test Plan:
python test/test_quantization.py TestQuantizePT2EQAT_ConvBn2d

Reviewers: jerryzh168, kimishpatel

Subscribers: jerryzh168, kimishpatel, supriyar

[ghstack-poisoned]

andrewor14 committed:
Update on "[quant][pt2][be] Refactor QAT tests for future patterns"


Summary: Currently the QAT tests are very specific to conv-bn-2d.
This makes it difficult to test new patterns like conv-bn-1d if
we want to add them. This commit refactors these tests so we can
add and test future patterns easily.

Test Plan:
python test/test_quantization.py TestQuantizePT2EQAT_ConvBn2d

Reviewers: jerryzh168, kimishpatel

Subscribers: jerryzh168, kimishpatel, supriyar

[ghstack-poisoned]

andrewor14 commented:
@pytorchbot merge
